tests/run/cdef_class_dataclass.pyx

```diff
@@ -195,9 +195,9 @@ cdef class TestFrozen:
     >>> inst = TestFrozen(a=5)
     >>> inst.a
     5.0
-    >>> inst.a = 2.
+    >>> inst.a = 2.  # doctest: +ELLIPSIS
     Traceback (most recent call last):
-    AttributeError: attribute 'a' of 'cdef_class_dataclass.TestFrozen' objects is not writable
+    AttributeError: attribute 'a' of '...TestFrozen' objects is not writable
```

Comment:
Sorry for the mess. What does PyPy print here?

Reply:
It's probably not just PyPy, is it? That's something that changed in Py3.10. It now puts the FQCN in exception messages.

Reply:
It looks like PyPy omits the module name in this case and most other versions of Python include it. I'm not convinced this is the sort of thing where PyPy is morally obliged to match exactly (although no doubt you have your own views on this) - it only matters because doctest matches the exact text.

Reply:
I think the rules of when the module name is expected in an error message are not clear. I remember going back and forth on this trying to get some project's (NumPy? Cython? stdlib?) tests to pass. The problem is something like pure-python classes are different from classes defined via the C-API or so.

Reply:
But if PyPy could improve, it can be done, just I need some direction.

Reply:
It looks to me like all the CPython versions from 2.7-3.10a6 produce `attribute 'attribute_name' of 'module_name.ClassName' objects is not writable`. (3.10a6 is just the highest version I have installed to test... I'm sure the current version is the same) I think in general CPython's trying to use the qualified (dotted) names everywhere, so their current behaviour is probably their future behaviour. So I think the most compatible behaviour would be for PyPy to all produce the qualified/dotted name.

For the sake of this test `#doctest: +ELLIPSIS` works perfectly well and tests everything I care about. So it'll probably remain.